js/main.js

```diff
@@ -55,6 +54,212 @@ function renderDate(obj) {
+  date = xhrMonth.response[0].date;
+  currentMonth = getMonth(date);
+
+  if (month[0].weekday !== 'sunday') {
+    if (month[0].weekday === 'monday') {
+      emptyDays = 1;
+    } if (month[0].weekday === 'tuesday') {
+      emptyDays = 2;
+    }
+    if (month[0].weekday === 'wednesday') {
+      emptyDays = 3;
+    }
+    if (month[0].weekday === 'thursday') {
+      emptyDays = 4;
+    }
+    if (month[0].weekday === 'friday') {
+      emptyDays = 5;
+    }
+    if (month[0].weekday === 'saturday') {
+      emptyDays = 6;
+    }
+    for (var i = 0; i < emptyDays; i++) {
+      var $p = document.createElement('p');
+      $p.className = 'hidden-cal';
+      $calendar.appendChild($p);
+    }
+  }
+  $calDate.textContent = currentMonth + ' 2022';
+  $monthTitle.textContent = currentMonth;
+  for (i = 0; i < month.length; i++) {
+    $p = document.createElement('p');
+    $p.setAttribute('id', i);
+    $p.className = 'cal rel';
+    $p.textContent = i + 1;
+    $calendar.appendChild($p);
+    if (month[i].celebrations[0].rank_num <= 2.8) {
+      var weekday = month[i].weekday;
+      var day = weekday.charAt(0).toUpperCase() + weekday.slice(1);
+      var $span = document.createElement('span');
+      $span.textContent = month[i].celebrations[0].title;
+      $p.appendChild($span);
+      var $li = document.createElement('li');
+      $li.textContent = day + ' ' + currentMonth + ' ' + (i + 1) + ', ' + month[i].celebrations[0].title;
+      $feastDayList.appendChild($li);
+    }
+  }
+  var lastItem = month.length - 1;
+  if (month[lastItem].weekday !== 'saturday') {
+    if (month[lastItem].weekday === 'friday') {
+      emptyDays = 1;
+    } if (month[lastItem].weekday === 'thursday') {
+      emptyDays = 2;
+    }
+    if (month[lastItem].weekday === 'wednesday') {
+      emptyDays = 3;
+    }
+    if (month[lastItem].weekday === 'tuesday') {
+      emptyDays = 4;
+    }
+    if (month[lastItem].weekday === 'monday') {
+      emptyDays = 5;
+    }
+    if (month[lastItem].weekday === 'sunday') {
+      emptyDays = 6;
+    }
+    for (i = 0; i < emptyDays; i++) {
+      $p = document.createElement('p');
+      $p.className = 'hidden-cal';
+      $calendar.appendChild($p);
+    }
+  }
+}
+
+function addEntry(event) {
+  event.preventDefault();
+  if (data.editing !== null) {
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.editing.id === data.entries[i].id) {
+        data.entries[i] = {
+          id: id,
+          color: color,
+          imageUrl: event.target.children[1].children[0].value,
+          notes: event.target.children[0].children[0].value
+        };
+        var $newEntry = createDomTree(xhrMonth.response[id]);
+        $journalPage.children[0].replaceWith($newEntry);
+        data.editing = null;
+        return;
+      }
+    }
+  }
+  var inputObj = {
+    id: id,
+    color: color,
+    imageUrl: event.target.children[1].children[0].value,
+    notes: event.target.children[0].children[0].value
```

Comment:
Rather than using indexes of the `children` array, you should use `querySelector` on `event.target` for a class or something on the element that identifies it.